src/app/clusters/basic/basic.cpp

```diff
@@ -318,6 +318,101 @@ void emberAfBasicClusterServerInitCallback(chip::EndpointId endpoint)
     }
 }
 
+EmberAfStatus BasicClusterServerGetVendorIdAttribute(chip::EndpointId endpointId, chip::VendorId & value)
+{
+    return Attributes::VendorID::Get(endpointId, &value);
+}
+
+EmberAfStatus BasicClusterServerSetVendorIdAttribute(chip::EndpointId endpointId, chip::VendorId value)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    chip::VendorId currentValue;
+    status = Attributes::VendorID::Get(endpointId, &currentValue);
+    VerifyOrDie(EMBER_ZCL_STATUS_SUCCESS == status);
+
+    if (currentValue != value)
```

Comment:
I'm not sure it's worth checking if the value is equal. Setting the same value again shouldn't be harmful. Especially that the attribute is stored in RAM only.